src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -71,26 +115,27 @@ void ScatterUpdate::getSupportedDescriptors() {
         OPENVINO_THROW(errorPrefix, " has incorrect number of input edges");
     if (getChildEdges().empty())
         OPENVINO_THROW(errorPrefix, " has incorrect number of output edges");
-
-    if (getInputShapeAtPort(DATA_ID).getRank() < 1 ||
-        getInputShapeAtPort(INDICES_ID).getRank() < 1 ||
-            getInputShapeAtPort(UPDATE_ID).getRank() < 1) {
-        OPENVINO_THROW(errorPrefix, " do not support scalar input");
-    }
-
+    // In ov::PartialShape with rank 0 (scalars) is converted to ov::intel_cpu::Shape with rank 1.
+    // Create workaround by extracting information about rank directly from OV node.
+    bool is_not_supported_input = ov::is_scalar(ovOp->get_input_partial_shape(0)) || ov::is_scalar(ovOp->get_input_partial_shape(1));
```

Comment:
Such validation should be done on the Core side. If you need to check static shapes in the Plugin, do that in ctr. That allows to switch to the Reference node. If dynamic - in the fn `prepareParams()`.

Reply:
Transferred checks to constructor and prepareParams

Reply:
Such check is a part of `shape_infer` for `ScatterNDBase`, so the input shapes will be validated during op creation.
Even if CPU plugin makes some internal shape adjustments, the shapes retrived by `op->get_input_partial_shape` should be the same as the shapes validated during `ScatterNDUpdate` constructor / validate_and_infer_types / shape_infer
https://github.com/openvinotoolkit/openvino/blob/dd50c29a8898820148d4bc1ef4f5f0fc45956e7c/src/core/shape_inference/include/scatter_nd_base_shape_inference.hpp#L25-L27
Is there any special reason to re-validate it here?

Reply:
Check for rank exists in current implementation of Scatter ops, so I thought that there might be some reasons behind it: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/scatter_update.cpp#L75-L79

However, if this is some kind of legacy that can be deleted, please let me know, I'll update it